sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ImplementISerializableCorrectly.cs

```diff
@@ -231,23 +229,24 @@ private static bool IsCallingBaseConstructor(IMethodSymbol constructorSymbol)
             return baseKeyword.HasValue && baseKeyword.Value.IsKind(SyntaxKind.BaseKeyword);
         }
 
-        private static bool ImplementsISerializable(ITypeSymbol typeSymbol)
-        {
-            return typeSymbol != null &&
-                typeSymbol.IsPubliclyAccessible() &&
-                typeSymbol.AllInterfaces.Any(IsOrImplementsISerializable);
-        }
+        private static bool ImplementsISerializable(ITypeSymbol typeSymbol) =>
+            typeSymbol != null &&
+            typeSymbol.IsPubliclyAccessible() &&
+            typeSymbol.AllInterfaces.Any(IsOrImplementsISerializable);
 
-        private static bool HasSerializableAttribute(ITypeSymbol typeSymbol)
-        {
-            return typeSymbol.GetAttributes()
+        private static bool HasSerializableAttribute(ITypeSymbol typeSymbol) =>
+            typeSymbol.GetAttributes()
                 .Any(a => a.AttributeClass.Is(KnownType.System_SerializableAttribute));
-        }
 
-        private static bool IsOrImplementsISerializable(ITypeSymbol typeSymbol)
-        {
-            return typeSymbol.Is(KnownType.System_Runtime_Serialization_ISerializable) ||
-                typeSymbol.Implements(KnownType.System_Runtime_Serialization_ISerializable);
-        }
+        private static bool IsOrImplementsISerializable(ITypeSymbol typeSymbol) =>
+            typeSymbol.Is(KnownType.System_Runtime_Serialization_ISerializable) ||
+            typeSymbol.Implements(KnownType.System_Runtime_Serialization_ISerializable);
+
+        private static bool IsPublicVirtual(IMethodSymbol methodSymbol) =>
+            methodSymbol.DeclaredAccessibility == Accessibility.Public &&
+            (methodSymbol.IsVirtual || methodSymbol.IsOverride);
+
+        private static bool IsExplicitImplementation(IMethodSymbol methodSymbol) =>
+            methodSymbol.Name == "System.Runtime.Serialization.ISerializable.GetObjectData";

```

Comment:
This is a bit odd. Would that work?
`methodSymbol.ExplicitInterfaceImplementations.Any(IsGetObjectData)`

I'd also rename the method to `IsExplicitImplementationOfGetObject` or something similar.